### PR TITLE
Fix MongoDB Manager stubs

### PR DESCRIPTION
**Pull request by GromNaN**
Sync some outdated stubs while [implementing rector on `mongodb/mongodb`](https://github.com/mongodb/mongo-php-library/pull/1088).

- `Server::executeBulkWrite` https://github.com/mongodb/mongo-php-driver/blob/b93f7c2bd10f6f1710aadb0505045df2d221b3ea/src/MongoDB/Server.stub.php#L74-L79

- `Manager::executeBulkWrite` https://github.com/mongodb/mongo-php-driver/blob/b93f7c2bd10f6f1710aadb0505045df2d221b3ea/src/MongoDB/Manager.stub.php#L18-L23

Some outdated definition might still exist, [I made a note to check it all later](https://jira.mongodb.org/browse/PHPORM-23).

### CODE REVIEW

File: resources/functionMap.php [hunk 1]
```diff
@@ -6774,11 +6774,12 @@
...
-'MongoDB\Driver\Manager::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Manager::addSubscriber' => ['void', 'subscriber'=>'MongoDB\Driver\Monitoring\Subscriber'],
+'MongoDB\Driver\Manager::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulk'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|null'],
+'MongoDB\Driver\Manager::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
 'MongoDB\Driver\Manager::executeDelete' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'filter'=>'array|object', 'deleteOptions='=>'array', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
 'MongoDB\Driver\Manager::executeInsert' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'document'=>'array|object', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
```
Comment (alcaeus) [inline resources/functionMap.php]: These two methods don't exist: https://www.php.net/manual/en/class.mongodb-driver-manager.php#mongodb-driver-manager.synopsis

File: resources/functionMap.php [hunk 2]
```diff
@@ -6774,11 +6774,12 @@
 'MongoDB\Driver\Exception\WriteException::getTraceAsString' => ['string'],
 'MongoDB\Driver\Exception\WriteException::getWriteResult' => ['MongoDB\Driver\WriteResult'],
 'MongoDB\Driver\Manager::__construct' => ['void', 'uri'=>'string', 'options='=>'array', 'driverOptions='=>'array'],
-'MongoDB\Driver\Manager::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulk'=>'MongoDB\Driver\BulkWrite', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
-'MongoDB\Driver\Manager::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Manager::addSubscriber' => ['void', 'subscriber'=>'MongoDB\Driver\Monitoring\Subscriber'],
+'MongoDB\Driver\Manager::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulk'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|null'],
+'MongoDB\Driver\Manager::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
```
Comment (alcaeus) [inline resources/functionMap.php:6781]: Note, there are also `executeReadCommand`, `executeReadWriteCommand`, and `executeWriteCommand`.

File: resources/functionMap.php [hunk 3]
```diff
@@ -6774,11 +6774,12 @@
...
 'MongoDB\Driver\Manager::executeDelete' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'filter'=>'array|object', 'deleteOptions='=>'array', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
 'MongoDB\Driver\Manager::executeInsert' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'document'=>'array|object', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
-'MongoDB\Driver\Manager::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Manager::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
 'MongoDB\Driver\Manager::executeUpdate' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'filter'=>'array|object', 'newObj'=>'array|object', 'updateOptions='=>'array', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
```
Comment (alcaeus) [inline resources/functionMap.php]: This method does not exist.

File: resources/functionMap.php [hunk 4]
```diff
@@ -6794,9 +6795,9 @@
 'MongoDB\Driver\ReadPreference::getMode' => ['int'],
 'MongoDB\Driver\ReadPreference::getTagSets' => ['array'],
 'MongoDB\Driver\Server::__construct' => ['void', 'host'=>'string', 'port'=>'string', 'options='=>'array', 'driverOptions='=>'array'],
-'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'zwrite'=>'BulkWrite'],
-'MongoDB\Driver\Server::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
-'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'bulkWrite'=>'BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|nullarray|MongoDB\Driver\WriteConcern|null'],
```
Comment (alcaeus) [inline resources/functionMap.php]: I think the `bulkWrite` argument needs a FQCN. There's also a missing return type

```suggestion
'MongoDB\Driver\Server::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulkWrite'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|null'],
```

File: resources/functionMap.php [hunk 5]
```diff
@@ -6794,9 +6795,9 @@
...
+'MongoDB\Driver\Server::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
+'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'options='=>'array|RMongoDB\Driver\ReadPreference|null'],
```
Comment (alcaeus) [inline resources/functionMap.php]: An extra letter snuck in here:
```suggestion
'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
```

**Review by alcaeus — CHANGES_REQUESTED** [review]

File: resources/functionMap.php (hunk 1 above)
Comment (GromNaN) [reply resources/functionMap.php]: Did they ever exist? I think we need a script update this file for the ext-mongodb reference.

Comment (alcaeus) [reply resources/functionMap.php]: No, these methods never existed.

File: resources/functionMap.php (hunk 2 above)
Comment (GromNaN) [reply resources/functionMap.php:6781]: Updated

File: resources/functionMap.php (hunk 4 above)
Comment (GromNaN) [reply resources/functionMap.php]: Fixed

File: resources/functionMap.php (hunk 3 above)
Comment (GromNaN) [reply resources/functionMap.php]: Removed

File: resources/functionMap.php (hunk 5 above)
Comment (GromNaN) [reply resources/functionMap.php]: Fixed

**Review by GromNaN — COMMENTED** [review]
Updated `MongoDB\Driver\Manager` using ext-mongodb source stubs: https://github.com/mongodb/mongo-php-driver/blob/master/src/MongoDB/Manager.stub.php

File: resources/functionMap.php [hunk 6]
```diff
@@ -6794,9 +6800,9 @@
 'MongoDB\Driver\ReadPreference::getMode' => ['int'],
 'MongoDB\Driver\ReadPreference::getTagSets' => ['array'],
 'MongoDB\Driver\Server::__construct' => ['void', 'host'=>'string', 'port'=>'string', 'options='=>'array', 'driverOptions='=>'array'],
-'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'zwrite'=>'BulkWrite'],
-'MongoDB\Driver\Server::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
-'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'bulkWrite'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|nullarray|MongoDB\Driver\WriteConcern|null'],
```
Comment (alcaeus) [inline resources/functionMap.php]: Return type is still missing, and the type for `options` is invalid:

```suggestion
'MongoDB\Driver\Server::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulkWrite'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|null'],
```

**Review by alcaeus — APPROVED** [review]

**Comment by GromNaN** [comment]
Replaced by https://github.com/phpstan/phpstan-src/pull/2473

**Pull request closed without merging**